libs/entities/src/entities/governance.delegated.vote.info.ts

```diff
@@ -0,0 +1,35 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { BinaryUtils } from '@multiversx/sdk-nestjs-common';
+import { VmQueryResponse } from '@libs/common/vm-query/dtos/vm-query.response';
+
+export class GovernanceDelegatedVoteInfo {
+  constructor(init?: Partial<GovernanceDelegatedVoteInfo>) {
+    Object.assign(this, init);
+  }
+
+  @ApiProperty({ description: 'The used stake for the delegated address.', example: '1000000000000000000' })
+  usedStake?: string;
+
+  @ApiProperty({ description: 'The used power for the delegated address.', example: '1000000000000000000' })
+  usedPower?: string;
+
+  @ApiProperty({ description: 'The total stake of the address.', example: '1000000000000000000' })
+  totalStake?: string;
+
+  @ApiProperty({ description: 'The address power of the address.', example: '1000000000000000000' })
+  totalPower?: string;
+
+  static fromVmQueryResponse(queryResponse: VmQueryResponse): GovernanceDelegatedVoteInfo {
+    const returnData = queryResponse?.data?.data?.returnData;
+    if (!returnData || !Array.isArray(returnData) || returnData.length === 0) {
```

Comment:
what happend if returnData.length === 3?

you can reduce the checks for only checking if the `( !returnData || returnData.length < 4 )` and you covered all the cases

Reply:
done